File: Back-End/src/main/java/vn/com/ecomstore/services/RankingService.java
package vn.com.ecomstore.services;

import vn.com.ecomstore.dtos.response.rank.RankResponse;
import vn.com.ecomstore.entities.Ranking;

import java.util.List;

public interface RankingService {
    Ranking getRankingEntityById(Long id);

    List<RankResponse> getAllRankings();
}

File: Front-End/src/services/ranking.service.ts
import axiosClient from "@/configurations/axios.config";
import type { RankResponse, Rank } from "@/types/ranking.type";

export const rankingService = {
  getAllRankings: async () => {
    const response = await axiosClient.get<RankResponse>("/rankings");
    return response.data;
  },
  getMyRank: async () => {
    const response = await axiosClient.get<{ status: number; message: string; data: Rank }>("/rankings/my-rank");
    return response.data;
  },
};
